File: eulerglot/ruby/lib/002_even_fibo_numbers.rb
class EvenFiboNumbers

  def self.sum_under(ceil)
    return 0  if ceil < 2
    return 2  if ceil < 10

    acc      = 0
    previous = 2
    current  = 8
    sum      = previous + current

    while current < ceil
      sum += acc
      acc = (4 * current) + previous
      previous = current
      current = acc
    end

    sum
  end

end

File: eulerglot/javascript/src/002EvenFiboNumbers.js
function sumUnder(ceil) {
  return sumFibo(1, 0, ceil, 0)
}

function sumFibo(current, previous, ceil, acc) {
  if (ceil <= 0) {
    return acc;
  }

  number = current + previous;

  if (number % 2 == 0) {
    return sumFibo(number, current, ceil - current, acc + number) ;
  } else {
    return sumFibo(number, current, ceil - current, acc);
  }
}


module.exports = {
  sumUnder: sumUnder,
  sumFibo: sumFibo
}
